File: 392-is-subsequence/is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0; // pointer for s
        int j = 0; // pointer for t

        while (i < s.size() && j < t.size()) {
            if (s[i] == t[j]) i++; // match found, move i
            j++; // always move j
        }

        return i == s.size(); // all chars in s were found in t in order
    }
};

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean Solve(int s_size, int t_size, String s, String t){
        if(s_size==0) return true;
        if(t_size==0) return false;
        if(s.charAt(s_size-1)==t.charAt(t_size-1)){
            return Solve(s_size-1,t_size-1,s,t);
        }
        else return Solve(s_size,t_size-1,s,t);
    }
    public boolean isSubsequence(String s, String t) {
        int s_size = s.length();
        int t_size = t.length();
        return Solve(s_size,t_size,s,t);
    }
}